modal/_serialization.py

```diff
@@ -42,13 +46,142 @@ def persistent_load(self, pid):
             raise InvalidError("bad flag")
 
 
-def serialize(obj):
+def serialize(obj: Any) -> bytes:
     """Serializes object and replaces all references to the client class by a placeholder."""
     buf = io.BytesIO()
     Pickler(buf).dump(obj)
     return buf.getvalue()
 
 
-def deserialize(s: bytes, client):
+def deserialize(s: bytes, client) -> Any:
     """Deserializes object and replaces all client placeholders by self."""
     return Unpickler(client, io.BytesIO(s)).load()
+
+
+def _serialize_asgi(obj: Any) -> api_pb2.Asgi:
+    def flatten_headers(obj):
+        return [s for k, v in obj for s in (k, v)]
+
+    msg_type = obj.get("type")
+    if msg_type == "http":
```

Comment:
are all of these covered by tests? presumably the container_test code covers most of these, but it's hard to know

Reply:
Okay sounds good! I'll round-trip a few requests through!